From 5dbcdc8ef4a20b4715cef65b253a003f2f3dc445 Mon Sep 17 00:00:00 2001
From: chris <chris.r.jones.1983@gmail.com>
Date: Wed, 6 Jul 2022 20:05:01 -0500
Subject: [PATCH] patch sbknumpyarrayconverter.cpp for shiboken2 with numpy
 v1.23

---
 sources/shiboken2/libshiboken/sbknumpyarrayconverter.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/sources/shiboken2/libshiboken/sbknumpyarrayconverter.cpp b/sources/shiboken2/libshiboken/sbknumpyarrayconverter.cpp
index 996968f..cc25b34 100644
--- a/sources/shiboken2/libshiboken/sbknumpyarrayconverter.cpp
+++ b/sources/shiboken2/libshiboken/sbknumpyarrayconverter.cpp
@@ -116,8 +116,13 @@ std::ostream &operator<<(std::ostream &str, PyArrayObject *o)
             str << " NPY_ARRAY_NOTSWAPPED";
-         if ((flags & NPY_ARRAY_WRITEABLE) != 0)
-             str << " NPY_ARRAY_WRITEABLE";
+#if NPY_VERSION >= 0x00000010 // NPY_1_23_API_VERSION
+        if ((flags & NPY_ARRAY_WRITEBACKIFCOPY) != 0)
+            str << " NPY_ARRAY_WRITEBACKIFCOPY";
+#else
+        if ((flags & NPY_ARRAY_UPDATEIFCOPY) != 0)
+            str << " NPY_ARRAY_UPDATEIFCOPY";
+#endif
     } else {
         str << '0';
     }
-- 
2.37.0

