diff --git a/src/MacAppBundle/CMakeLists.txt b/src/MacAppBundle/CMakeLists.txt
index c0b6ccf09f..b64cd761bf 100644
--- a/src/MacAppBundle/CMakeLists.txt
+++ b/src/MacAppBundle/CMakeLists.txt
@@ -112,16 +112,16 @@ install(CODE "execute_process(COMMAND chmod -R a+w ${CMAKE_INSTALL_LIBDIR})")
 get_filename_component(APP_PATH ${CMAKE_INSTALL_PREFIX} PATH)

 execute_process(
-       COMMAND find /usr/local/Cellar/icu4c -name pkgconfig
-       RESULT_VARIABLE CMD_ERROR
-       OUTPUT_VARIABLE CONFIG_ICU)
+    COMMAND find ${HOMEBREW_PREFIX}/Cellar/icu4c -name pkgconfig
+    RESULT_VARIABLE CMD_ERROR
+    OUTPUT_VARIABLE CONFIG_ICU)
 set(ENV{PKG_CONFIG_PATH} "$ENV{PKG_CONFIG_PATH}:${CONFIG_ICU}")
 find_package(PkgConfig)
 pkg_check_modules(ICU icu-uc)

 execute_process(
-       COMMAND find /usr/local/Cellar/nglib -name MacOS
-       OUTPUT_VARIABLE CONFIG_NGLIB)
+    COMMAND find ${HOMEBREW_PREFIX}/Cellar/nglib@6.2.2104 -name MacOS
+    OUTPUT_VARIABLE CONFIG_NGLIB)

 install(CODE
     "message(STATUS \"Making bundle relocatable...\")
@@ -130,6 +130,6 @@ install(CODE
     execute_process(
         COMMAND python2.7
         ${CMAKE_SOURCE_DIR}/src/Tools/MakeMacBundleRelocatable.py
-        ${APP_PATH} ${HOMEBREW_PREFIX}${MACPORTS_PREFIX}/lib ${ICU_PREFIX}/lib/ /usr/local/opt ${CONFIG_NGLIB} ${Qt5Core_DIR}/../../.. ${XCTEST_PATH} ${WEBKIT_FRAMEWORK_DIR}
+        ${APP_PATH} ${HOMEBREW_PREFIX}${MACPORTS_PREFIX}/lib ${HOMEBREW_PREFIX}/opt/gcc/lib/gcc/11 ${HOMEBREW_PREFIX}/opt/llvm@11/lib ${HOMEBREW_PREFIX}/opt/pyside2@5.15.2/lib/ ${HOMEBREW_PREFIX}/opt/icu4c/lib/ ${CONFIG_NGLIB} ${HOMEBREW_PREFIX}/opt ${HOMEBREW_PREFIX}/opt/opencascade@7.5.3/lib ${Qt5Core_DIR}/../../.. ${XCTEST_PATH} ${WEBKIT_FRAMEWORK_DIR} /usr/lib
     )"
 )
