From 97ada4137d69fe17ff15569e1d7d2a146aa78ccf Mon Sep 17 00:00:00 2001
From: chris <chris.r.jones.1983@gmail.com>
Date: Mon, 22 Aug 2022 16:06:20 -0500
Subject: [PATCH] macos fixes for mac app bundling

---
 src/MacAppBundle/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/MacAppBundle/CMakeLists.txt b/src/MacAppBundle/CMakeLists.txt
index 07876170dec0..3dfb655d6bac 100644
--- a/src/MacAppBundle/CMakeLists.txt
+++ b/src/MacAppBundle/CMakeLists.txt
@@ -114,7 +114,7 @@ install(CODE "execute_process(COMMAND chmod -R a+w ${CMAKE_INSTALL_LIBDIR})")
 get_filename_component(APP_PATH ${CMAKE_INSTALL_PREFIX} PATH)
 
 execute_process(
-    COMMAND find /usr/local/Cellar/icu4c -name pkgconfig
+    COMMAND find ${HOMEBREW_PREFIX}/Cellar/icu4c -name pkgconfig
     RESULT_VARIABLE CMD_ERROR
     OUTPUT_VARIABLE CONFIG_ICU)
 set(ENV{PKG_CONFIG_PATH} "$ENV{PKG_CONFIG_PATH}:${CONFIG_ICU}")
@@ -132,6 +132,6 @@ install(CODE
     execute_process(
         COMMAND python3
         ${CMAKE_SOURCE_DIR}/src/Tools/MakeMacBundleRelocatable.py
-        ${APP_PATH} ${HOMEBREW_PREFIX}${MACPORTS_PREFIX}/lib ${ICU_PREFIX}/lib/ /usr/local/opt ${CONFIG_NGLIB} ${Qt5Core_DIR}/../../.. ${XCTEST_PATH} ${WEBKIT_FRAMEWORK_DIR}
+        ${APP_PATH} ${HOMEBREW_PREFIX}${MACPORTS_PREFIX}/lib ${HOMEBREW_PREFIX}/opt/icu4c/lib ${HOMEBREW_PREFIX}/opt/llvm/lib ${HOMEBREW_PREFIX}/opt/med-file@4.1.1/lib ${HOMEBREW_PREFIX}/opt/gcc/lib/gcc/12 /usr/local/opt ${CONFIG_NGLIB} ${Qt5Core_DIR}/../../.. ${XCTEST_PATH} ${WEBKIT_FRAMEWORK_DIR}
     )"
 )
